packages/machine-extractor/src/extractAction.ts

```diff
@@ -0,0 +1,411 @@
+    if (t.isObjectExpression(arg)) {
+      return extractEventObject(arg, fileContent);
+    }
+
+    // raise('event')
+    if (t.isStringLiteral(arg)) {
+      return { type: { type: 'string', value: arg.value } };
+    }
+  }
+
+  throw Error(`Unsupported raised event`);
+}
+
+export function extractLogExpression(
+  actionNode: ActionNode,
+  fileContent: string,
+): { type: JsonEntryType; value: JsonValue } {
+  const node = actionNode.node;
+  if (t.isCallExpression(node)) {
+    const arg = node.arguments[0];
+
+    // log('string')
+    if (t.isStringLiteral(arg)) {
+      return {
+        type: 'string',
+        value: arg.value,
+      };
+    }
+
+    // log((ctx, evt) => {})
+    if (t.isArrowFunctionExpression(arg) || t.isFunctionExpression(arg)) {
+      return {
+        type: 'expression',
+        value: fileContent.slice(node.start!, node.end!),
+      };
+    }
+  }
+
+  throw Error(`Unsupported log expression`);
+}
+
+export function extractStopProperties(
+  actionNode: ActionNode,
+  fileContent: string,
+): { type: JsonEntryType; value: JsonValue } {
+  const node = actionNode.node;
+  if (t.isCallExpression(node)) {
+    const arg = node.arguments[0];
+
+    if (t.isStringLiteral(arg)) {
+      return { type: 'string', value: arg.value };
+    }
+
+    if (t.isArrowFunctionExpression(arg) || t.isFunctionExpression(arg)) {
+      return {
+        type: 'expression',
+        value: fileContent.slice(arg.start!, arg.end!),
+      };
+    }
+  }
+
+  throw Error('Unsupported stop expression');
+}
+
+export function extractSendToProperties(
+  actionNode: ActionNode,
+  fileContent: string,
+): {
+  event:
+    | Record<
+        string,
+        {
+          type: JsonEntryType;
+          value: JsonValue;
+        }
+      >
+    | { type: 'expression'; value: string };
+  actorRef: {
+    type: JsonEntryType;
+    value: JsonValue;
+  };
+  options: {
+    delay: { type: JsonEntryType; value: JsonValue } | undefined;
+    id: { type: JsonEntryType; value: JsonValue } | undefined;
+  };
+} {
+  const node = actionNode.node;
+  let eventObject:
+    | Record<
+        string,
+        {
+          type: JsonEntryType;
+          value: JsonValue;
+        }
+      >
+    | { type: 'expression'; value: string } = {};
+  let actorRef:
+    | {
+        type: JsonEntryType;
+        value: JsonValue;
+      }
+    | undefined = undefined;
+  let delay: { type: JsonEntryType; value: JsonValue } | undefined = undefined;
+  let id: { type: JsonEntryType; value: JsonValue } | undefined = undefined;
+
+  if (t.isCallExpression(node)) {
+    const arg1 = node.arguments[0];
+    const arg2 = node.arguments[1];
+    const arg3 = node.arguments[2];
+
+    // Actor
+    // sendTo('actorName')
+    if (t.isStringLiteral(arg1)) {
+      actorRef = { type: 'string', value: arg1.value };
+    }
+    // sendTo((ctx, e) => actorRef)
+    else if (
+      t.isArrowFunctionExpression(arg1) ||
+      t.isFunctionExpression(arg1)
+    ) {
```

Comment:
why do we special case functions here? what about `sendTo(someActorRef, { type: 'FOO' })`?